symengine/tests/basic/test_arit.cpp

```diff
@@ -65,85 +65,85 @@ TEST_CASE("Add: arit", "[arit]")
     RCP<const Basic> r1 = add(x, x);
     RCP<const Basic> r2 = mul(i2, x);
     RCP<const Basic> r3 = mul(i3, x);
-    REQUIRE(eq(*r1, *r2));
-    REQUIRE(neq(*r1, *r3));
+    CHECK(eq(*r1, *r2));
+    CHECK(neq(*r1, *r3));
 
     r3 = mul(i2, y);
-    REQUIRE(neq(*r1, *r3));
-    REQUIRE(neq(*r2, *r3));
+    CHECK(neq(*r1, *r3));
+    CHECK(neq(*r2, *r3));
 
     r1 = add(mul(y, x), mul(mul(i2, x), y));
     r2 = mul(mul(i3, x), y);
-    REQUIRE(eq(*r1, *r2));
+    CHECK(eq(*r1, *r2));
 
     r1 = add({mul(y, x), mul({i2, x, y})});
     r2 = mul({i3, x, y});
-    REQUIRE(eq(*r1, *r2));
+    CHECK(eq(*r1, *r2));
 
     r1 = add(add(x, x), x);
     r2 = mul(i3, x);
-    REQUIRE(eq(*r1, *r2));
+    CHECK(eq(*r1, *r2));
 
     r1 = add(add(x, x), x);
     r2 = mul(x, i3);
-    REQUIRE(eq(*r1, *r2));
+    CHECK(eq(*r1, *r2));
     r1 = add({x, x, x});
-    REQUIRE(eq(*r1, *r2));
+    CHECK(eq(*r1, *r2));
 
     r1 = add(x, one);
     r2 = add(one, x);
-    REQUIRE(eq(*r1, *r2));
+    CHECK(eq(*r1, *r2));
 
     r1 = add(pow(x, y), z);
     r2 = add(z, pow(x, y));
-    REQUIRE(eq(*r1, *r2));
+    CHECK(eq(*r1, *r2));
     r2 = add({z, pow(x, y)});
-    REQUIRE(eq(*r1, *r2));
+    CHECK(eq(*r1, *r2));
 
     r1 = add(x, I);
     r2 = add(I, x);
-    REQUIRE(eq(*r1, *r2));
+    CHECK(eq(*r1, *r2));
     r2 = add({I, x});
-    REQUIRE(eq(*r1, *r2));
+    CHECK(eq(*r1, *r2));
 
     r1 = mul(x, I);
     r2 = mul(mul(I, i2), x);
     r3 = mul(mul(I, i3), x);
     r2 = add(r1, r2);
-    REQUIRE(eq(*r3, *r2));
+    CHECK(eq(*r3, *r2));
 
     r1 = real_double(0.1);
     r2 = Rational::from_mpq(rational_class(1, 2));
     r3 = add(add(add(r1, r2), integer(1)), real_double(0.2));
-    REQUIRE(is_a<RealDouble>(*r3));
-    REQUIRE(std::abs(down_cast<const RealDouble &>(*r3).i - 1.8) < 1e-12);
+    CHECK(is_a<RealDouble>(*r3));
```

Comment:
This is not correct. If this check fails the check below will crash. This has to be a `REQUIRE`